app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,18 @@
+class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  def facebook
```

Comment:
To be clear, omniauth_callbacks_controller isn't a controller for a resource in your project. That's why it could use a non-CRUD related action names here. I wrote this comment just so you will have a complete picture of what I discussed in the other controllers.